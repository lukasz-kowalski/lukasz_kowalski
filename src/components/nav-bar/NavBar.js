import React from 'react';
import { NavLink } from 'react-router-dom';

class NavBar extends React.Component {
  state = {
    isOpen: false
  };

  onToggle = () => {
    this.setState({
      isOpen: !this.state.isOpen
    });
  }

  render() {
    return (
      <nav className="navbar navbar-expand-sm navbar-dark bg-primary mb-3">
        <NavLink to="/" exact className="navbar-brand">Sonalake Task</NavLink>
        <button 
          type="button"
          onClick={this.onToggle}
          className={'navbar-toggler' + (!this.state.isOpen ? ' collapsed' : '')}
          aria-expanded={this.state.isOpen}
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon"/>
        </button>

        <div className={'collapse navbar-collapse' + (this.state.isOpen ? ' show' : '')}>
          <ul className=" navbar-nav">
            <li className=" nav-item active">
              <NavLink to="/" exact className="nav-link">
                List View
                <span className=" sr-only">(current)</span>
              </NavLink>
            </li>
          </ul>
        </div>
      </nav>
    ); 
  }
}

export default NavBar;
