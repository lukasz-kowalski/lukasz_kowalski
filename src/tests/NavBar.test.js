import React from 'react';
import { shallow } from 'enzyme';
import NavBar from '../components/nav-bar/NavBar';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<NavBar />);
})

test('should successful render NavBar', () => {
  expect(wrapper).toMatchSnapshot();
})

test('should change component state isOpen to true after first click on button', () => {
  wrapper.find('.navbar-toggler').simulate('click');
  expect(wrapper.state('isOpen')).toBe(true);
})

test('should change component state isOpen to false after second click on button', () => {
  wrapper.find('.navbar-toggler').simulate('click');
  wrapper.find('.navbar-toggler').simulate('click');
  expect(wrapper.state('isOpen')).toBe(false);
})
