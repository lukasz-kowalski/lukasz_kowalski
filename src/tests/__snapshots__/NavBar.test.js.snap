// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successful render NavBar 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-primary mb-3"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="navbar-brand"
    exact={true}
    to="/"
  >
    Sonalake Task
  </NavLink>
  <button
    aria-expanded={false}
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
  >
    <ul
      className=" navbar-nav"
    >
      <li
        className=" nav-item active"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link"
          exact={true}
          to="/"
        >
          List View
          <span
            className=" sr-only"
          >
            (current)
          </span>
        </NavLink>
      </li>
    </ul>
  </div>
</nav>
`;
